xcom2-launcher/xcom2-launcher/Classes/Log4Net/SentryAppender.cs

```diff
@@ -0,0 +1,215 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using log4net.Appender;
+using log4net.Core;
+using Sentry;
+using Sentry.Extensibility;
+using Sentry.Protocol;
+
+namespace XCOM2Launcher.Log4Net
+{
+    /// <summary>
+    /// Sentry appender for log4net.
+    /// </summary>
+    public class SentryAppender : AppenderSkeleton
+    {
+        private readonly Func<string, IDisposable> _initAction;
+        private volatile IDisposable _sdkHandle;
+
+        private readonly object _initSync = new object();
+
+        internal static readonly SdkVersion NameAndVersion
+            = typeof(SentryAppender).Assembly.GetNameAndVersion();
+
+        private static readonly string ProtocolPackageName = NameAndVersion.Name;
+
+        internal IHub Hub { get; set; }
+
+        /// <summary>
+        /// Sentry DSN.
+        /// </summary>
+        public string Dsn { get; set; }
```

Comment:
What's the purpose of this, especially if you don't set it in the config?

Reply:
Its just an option to specify the Dsn in log4net.config instead of using Sentry.Ini(). I kept it generic because I will potentially create a related PR for the Sentry project.